ast/ast_test.go

```diff
@@ -0,0 +1,375 @@
+}
+
+func TestWriteASTToFile(t *testing.T) {
+	config := `; Some preamble
+[hallMaRk]
+christmas = merry
+newyear = happy
+
+[Rosaceae]
+Malus domestica = Orchard apple
+`
+	// Read config into an ast.File
+	file := Read(strings.NewReader(config), "test")
+	Write(file, file.Name)
+	defer os.Remove(file.Name)
+
+	if err := os.WriteFile("expected", []byte(config), 0644); err != nil {
+		t.Errorf("Error writing file to disk")
+	}
+	defer os.Remove("expected")
+
+	require.True(t, deepCompare("test", "expected"), string(convertASTToBytes(file)))
+}
+
+func TestReadInjectWrite(t *testing.T) {
+	config := `[hallMaRk]
+christmas = merry
+newyear = happy
+
+[Rosaceae]
+Malus domestica = Orchard apple
+`
+	file := Read(strings.NewReader(config), "test")
+	field := Field{
+		Key:   "Malus prunifolia",
+		Value: "Chinese crabapple",
+	}
+	file = InjectField(file, field, "rosaceae", true)
+	Write(file, file.Name)
+	defer os.Remove(file.Name)
+
+	expectedResult := `[hallMaRk]
+christmas = merry
+newyear = happy
+
+[Rosaceae]
+Malus domestica = Orchard apple
+Malus prunifolia = Chinese crabapple
+`
+	if err := os.WriteFile("expected", []byte(expectedResult), 0644); err != nil {
+		t.Errorf("Error writing file to disk")
+	}
+	defer os.Remove("expected")
+
+	require.True(t, deepCompare("test", "expected"))
+}
+
+func TestHandleComments(t *testing.T) {
+	config := `[hallMaRk]
+christmas = merry
+newyear = happy
+
+[Rosaceae]
+; Malus is a genus of small deciduous
+; trees in the Rosaceae family
+Malus domestica = Orchard apple
+`
+	file := Read(strings.NewReader(config), "test")
+	require.Equal(t, len(file.Sections[1].Fields), 3)
+
+	field := Field{
+		Key:   "Malus prunifolia",
+		Value: "Chinese crabapple",
+	}
+	file = InjectField(file, field, "rosaceae", true)
+	Write(file, file.Name)
+	defer os.Remove(file.Name)
+
+	expectedResult := `[hallMaRk]
+christmas = merry
+newyear = happy
+
+[Rosaceae]
+; Malus is a genus of small deciduous
+; trees in the Rosaceae family
+Malus domestica = Orchard apple
+Malus prunifolia = Chinese crabapple
+`
+	if err := os.WriteFile("expected", []byte(expectedResult), 0644); err != nil {
+		t.Errorf("Error writing file to disk")
+	}
+	defer os.Remove("expected")
+
+	require.Equal(t, len(file.Sections[1].Fields), 4)
+	require.True(t, deepCompare("test", "expected"))
+}
+
+func TestHandleSubsections(t *testing.T) {
+	config := `[hallMaRk]
+christmas = merry
+newyear = happy
+
+[Rosaceae "subsection"]
+; Malus is a genus of small deciduous
+; trees in the Rosaceae family
+Malus domestica = Orchard apple
+`
+	file := Read(strings.NewReader(config), "test")
+	require.Equal(t, len(file.Sections), 2)
+	require.Equal(t, file.Sections[1].Key, "rosaceae \"subsection\"")
+	require.Equal(t, file.Sections[1].Header, "[Rosaceae \"subsection\"]")
+
+	field := Field{
+		Key:   "Malus prunifolia",
+		Value: "Chinese crabapple",
+	}
+	file = InjectField(file, field, "rosaceae \"subsection\"", true)
```

Comment:
From a public API perspective, I think it's preferable to offer something like `InjectField(file, field, "rosaceae", "subsection", true)`, then inside the function we can construct it as we wish.